roles/openshift_hosted/tasks/registry/storage/s3.yml

```diff
@@ -0,0 +1,105 @@
+---
+- fail:
+    msg: >
+      openshift_hosted_registry_storage_s3_accesskey and
+      openshift_hosted_registry_storage_s3_secretkey are required
+  when: openshift.hosted.registry.storage.s3.accesskey | default(none) is none or openshift.hosted.registry.storage.s3.secretkey | default(none) is none
+
+- fail:
+    msg: >
+      openshift_hosted_registry_storage_s3_bucket and
+      openshift_hosted_registry_storage_s3_region are required
+  when: openshift.hosted.registry.storage.s3.bucket | default(none) is none or openshift.hosted.registry.storage.s3.region | default(none) is none
+
+# Currently this is relying on the boto config (or env vars) on the master host, it
+# should be a local_action and use separate credentials for creating the
+# bucket (vs the ones for accessing the bucket)
+- name: Create the S3 bucket
+  become: no
+  local_action:
+    module: s3
+    bucket: "{{ openshift.hosted.registry.storage.s3.bucket }}"
+    region: "{{ openshift.hosted.registry.storage.s3.region }}"
+    mode: create
+  when: openshift.hosted.registry.storage.s3.create | default(false) | bool
+
```

Comment:
Everything below this point is most likely common to all object storage configuration and not just s3.